celery_serverless/worker_management.py

```diff
@@ -71,13 +72,14 @@ def attach_hooks(wait_connection=4.0, wait_job=1.0):
     Register the needed hooks:
     - One to trigger stop getting tasks after the 1st, and shutdown when done
     """
-    logger.info('Attaching hooks')
+    logger.info('Attaching Celery hooks')
     logger.debug('Wait connection time: %.2f', wait_connection)
     logger.debug('Wait job time: %.2f', wait_job)
 
     @celeryd_init.connect  # After worker process up
     def _broker_connection_timeout(*args, **kwargs):
-        return wakeme_soon(reason='broker connection', delay=wait_connection, callback=shutdown_when_done)
+        _shutdown = partial(shutdown_when_done, reason='Broker never connected')
+        return wakeme_soon(reason='broker connection', delay=wait_connection, callback=_shutdown)
```

Comment:
Consider `wakeme_soon` with `callback_args` and `callback_kwargs` as extra parameters to be passed to `wakeme_soon`.